src/components/DataDrawer/DataTable/index.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { useState, useEffect } from 'react';
+import Papa from 'papaparse';
+import { useSelector } from 'react-redux';
+import {
+  Table,
+  TableBody,
+  TableContainer,
+  TableHead,
+  makeStyles,
+  Paper,
+} from '@material-ui/core';
+import { getCurrTable } from '../../../context/tableStateSlice';
+import DataTableRow from './DataTableRow';
+
+const useStyles = makeStyles({
+  root: {
+    width: '100%',
+    height: '70vh',
+    color: 'black',
+  },
+  container: {
+    maxHeight: '100%',
+  },
+  headCells: {
+    color: 'black',
+  },
+  tableCells: {
+    color: 'darkGrey',
+  },
+});
+
+export interface DataTableConfig {
+  maxResults: number;
+}
+
+const DataTable = ({ maxResults }: DataTableConfig) => {
+  const classes = useStyles();
+
+  // get and destructure the currently open table
+  const { title, table, legendText } = useSelector(getCurrTable);
+  // set up state to store results from parsing the CSV
+  const [tableJson, setTableJson] = useState<any[]>([]);
+
+  const tableUrl = process.env.PUBLIC_URL + table;
+
+  // parse the csv, but only when we get a new table to parse
+  useEffect(() => {
```

Comment:
I'd move this async logic to the reducer that sets the currently active table, then save the results of the async call in Redux as a separate action, and have this component pull that state from Redux, with a loading display to take care of the delay before the async data is ready.